src/sim-host/protocol/socket.js

```diff
@@ -80,10 +83,21 @@ module.exports.initialize = function (pluginHandlers, services) {
     socket.on('refresh', function () {
         document.location.reload(true);
     });
+    socket.on('connect', function () {
+        deferred.resolve();
+    });
+    socket.on('connect_error', function (err) {
+        deferred.reject(err);
+    });
+    socket.on('connect_timeout', function (err) {
+        deferred.reject(err);
+    });
```

Comment:
This is just to make the function return a promise so it can be correctly chained with other promises. It shouldn't impact anything, as consumers can still ignore the returned promise if they want this call to happen purely asynchronously.